clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -270,12 +262,12 @@ public boolean partitionsAutoAssigned() {
         return !this.subscription.isEmpty();
     }
 
-    public void consumed(TopicPartition tp, long offset) {
-        assignedState(tp).consumed(offset);
+    public void position(TopicPartition tp, long offset) {
```

Comment:
These kind of cases are probably best left to be handled in user code. The user can collect the messages in their own batch until they have enough data to process. Until we have a really compelling use case, I think I prefer the simpler approach in this patch since maintaining `consumed` and `fetched` has proven error-prone. 

There is one unfortunate side effect of this change which doesn't appear to impact current code, but should mentioned anyway. If you call `initFetches` while you have records available in the internal buffer, you will end up sending an unnecessary fetch. I'm checking if there's a simple way to fix this problem.